src/handler/global/settings.ts

```diff
@@ -13,57 +15,132 @@ import {
     Toggles,
 } from "../../repository/UserSettingsRepository";
 
-export const data = new SlashCommandBuilder()
-    .setName("settings")
-    .setDescription("View and toggle settings for notifications");
+import { ButtonColor } from "../../service/util/constants";
 
-enum Color {
-    PRIMARY = "PRIMARY",
-    SECONDARY = "SECONDARY",
-}
+export class Settings {
+    @CommandData({ type: "global" })
+    settingsData() {
+        return new SlashCommandBuilder()
+            .setName("settings")
+            .setDescription("View and toggle settings for notifications")
+            .toJSON();
+    }
 
-function getColorFromBoolean(bool: boolean): Color {
-    if (bool) {
-        return Color.PRIMARY;
-    } else {
-        return Color.SECONDARY;
+    @Command({ name: "settings" })
+    @Guard({ body: commandChannelGuard })
```

Comment:
shouldn't it be either restricted entirely to DMs, or only use ephemeral replies? Seems weird to have this in the command channels. Also, can't other people press the buttons in that case?